### DISCUSSION TITLE
ImageRequest not supported for local drawables?

### DISCUSSION THREAD

**Discussion by ivanovpv** [coil-kt/coil#1068] [General]
I'm trying to use following code:

```
val drawable = when {
        mimeType.startsWith("image") -> R.drawable.ic_baseline_image_24
        mimeType.startsWith("video") -> R.drawable.ic_baseline_ondemand_video_24
        mimeType.startsWith("audio") -> R.drawable.ic_baseline_audiotrack_24
        else -> R.drawable.ic_baseline_attach_file_24
}
val imageRequest = ImageRequest.Builder(context)
        .data(drawable)
        .crossfade(true)
        .target(imageView)
        .build()

//blah-blah
// and somewhere deep inside code
imageLoader.enqueue(imageRequest)
```

But it doesn't work. I suspect that `.data(R.drawable.XXX)` is unworkable. Am I right or I miss something else?

**Comment by colinrtwhite** [comment]
This should work. What's the stacktrace for the error you're getting?

**Reply by ivanovpv** [reply]
I figured issue. Problem was in creation of `ImageRequest` with `Application` `Context` rather than `Activity` `Context`. I think it would be useful to mention it in docs - I mean usage of application context is wrong.
